File: 1521-find-a-value-of-a-mysterious-function-closest-to-target/1521-find-a-value-of-a-mysterious-function-closest-to-target.java
class Solution {
    public int closestToTarget(int[] arr, int target) {
        HashSet<Integer> s = new HashSet<>();
        int res = Integer.MAX_VALUE;
        for (int i = 0; i < arr.length; ++i) {
            HashSet<Integer> s1 = new HashSet<>();
            for (int x : s) {
                s1.add(x & arr[i]);
            }
            s1.add(arr[i]);
            for (int x : s1) {
                res = Math.min(res, Math.abs(x - target));
            }
            s = s1;
        }
        return res;
    }
}

File: 1521-find-a-value-of-a-mysterious-function-closest-to-target/1521-find-a-value-of-a-mysterious-function-closest-to-target.ts
function closestToTarget(arr: number[], target: number): number {
    let s = new Set<number>();
    let res = Number.MAX_VALUE;
    for (let i = 0; i < arr.length; ++i) {
        let s1 = new Set<number>();
        for (let x of s) {
            s1.add(x & arr[i]);
        }
        s1.add(arr[i]);
        for (let x of s1) {
            res = Math.min(res, Math.abs(x - target));
        }
        s = s1;
    }
    return res;
};